mkosi

```diff
@@ -2640,6 +2667,10 @@ def run_build_script(args, workspace, raw):
         else:
             cmdline.append("--chdir=/root")
 
+        if args.build_dir is not None:
+            cmdline.append("--setenv=BUILDDIR=/root/build")
```

Comment:
> Hence, what would mkosi do? force things to be out-of-tree? for things to be in-tree then? I am sure we'd break half of the existing projects by either choice...

Indeed, that sounds like a good argument against setting it unconditionally. Maybe this should be documented in the README though. Appending "`$BUILDDIR` will not be set if this directory is not present" or so to make it explicit should be enough I think.